core/Rename.hpp

```diff
@@ -55,6 +55,24 @@ namespace olympia
 
         //! \brief Name of this resource. Required by sparta::UnitFactory
         static const char name[];
+        ~Rename(){
+            // delete map_table and reference counter, as they are dynamically allocated
+            for(int i = 0; i < core_types::N_REGFILES; ++i){
+                delete [] map_table[i];
+                delete [] reference_counter[i];
+            }
+        }
+        // map of ARF -> PRF
+        std::queue<unsigned int> ** map_table = new std::queue<unsigned int>*[core_types::N_REGFILES];
+        // reference counter for PRF
+        int* reference_counter[core_types::N_REGFILES];
+        // list of free PRF that are available to map
+        std::queue<unsigned int> freelist[core_types::N_REGFILES];
+        
+        std::map<olympia::core_types::RegFile, unsigned int> NumRenameRegisters = {
+            {core_types::RegFile::RF_INTEGER, 512},
+            {core_types::RegFile::RF_FLOAT, 512}
+        };
```

Comment:
A few requested changes here.

1. These variables need to be in the `private` section of the class.  Need to maintain encapsulation.
2. The variable names are a mixture of different styles.  Please rename them in the style `reference_counter_` -- words separated by an `_` and end with an `_` since they are (will be) private
3. You should _never_ have "naked" pointers initialized with a raw `new`.  Even with a destructor, this is a memory leak waiting to happen (esp if an exception were thrown during construction).  _Always_ use `std::unique_ptr`.
4. The number of rename registers _per register file_ should come from parameters and not hard-coded

Reply:
If you need public access to these variables for your tester, consider creating a RenameTester class that is a `friend` of Rename so you can access the variables as private variables.

Reply:
Addressed in [163a70b](https://github.com/riscv-software-src/riscv-perf-model/pull/40/commits/163a70b5862658deca0df16b035bbe59dc841c92)